ddtrace/llmobs/_integrations/langchain.py

```diff
@@ -20,6 +28,114 @@
+        if not self.llmobs_enabled:
+            return
+        model_provider = span.get_tag(PROVIDER)
+        span.set_tag_str(SPAN_KIND, "llm")
+        span.set_tag_str(MODEL_NAME, span.get_tag(MODEL) or "")
+        span.set_tag_str(MODEL_PROVIDER, model_provider or "")
+
+        # input parameters
+        if model_provider:
+            input_parameters = {}
+            temperature = span.get_tag(f"langchain.request.{model_provider}.parameters.temperature")
+            max_tokens = span.get_tag(f"langchain.request.{model_provider}.parameters.max_tokens")
+            if temperature:
+                input_parameters["temperature"] = float(temperature)
+            if max_tokens:
+                input_parameters["max_tokens"] = int(max_tokens)
+            if input_parameters:
+                span.set_tag_str(INPUT_PARAMETERS, json.dumps(input_parameters))
+
+        if operation == "llm":
+            self._llmobs_set_meta_tags_from_llm(span, inputs, response, error)
+        elif operation == "chat_model":
+            self._llmobs_set_meta_tags_from_chat_model(span, inputs, response, error)
+        elif operation == "chain":
+            # TODO to be added as a follow-up PR
+            pass
+        self._llmobs_set_metrics_tags(span)
+
+    def _llmobs_set_meta_tags_from_llm(
+        self,
+        span: Span,
+        prompts: List[Any],
+        completions: Any,
+        err: bool = False,
+    ) -> None:
+        # input messages
+        if isinstance(prompts, str):
+            prompts = [prompts]
+        span.set_tag_str(INPUT_MESSAGES, json.dumps([{"content": str(prompt)} for prompt in prompts]))
+
+        # output messages
+        message_content = [{"content": ""}]
+        if not err:
+            message_content = [{"content": self.trunc(completion[0].text)} for completion in completions.generations]
+        span.set_tag_str(OUTPUT_MESSAGES, json.dumps(message_content))
+
+    def _llmobs_set_meta_tags_from_chat_model(
+        self,
+        span: Span,
+        chat_messages: List[List[Any]],
+        chat_completions: Any,
+        err: bool = False,
+    ) -> None:
+        # input messages
+        input_messages = []
+        for message_set in chat_messages:
+            for message in message_set:
+                input_messages.append(
+                    {
+                        "content": self.trunc(message.content),
+                        "role": str(message.type),
+                    }
+                )
+        span.set_tag_str(INPUT_MESSAGES, json.dumps(input_messages))
+
+        # output messages
+        output_messages = [{"content": ""}]
+        if not err:
+            output_messages = []
+            for message_set in chat_completions.generations:
+                for chat_completion in message_set:
+                    output_messages.append(
+                        {
+                            "content": self.trunc(chat_completion.text),
```

Comment:
No truncation here as well, also same concerns about type checking and checking for the `type` field